src/language_concepts/generics.md

```diff
@@ -0,0 +1,34 @@
+# Generics
+
+Generics allow you to use the same functions with multiple different concrete data types. You can read more about the concept of generics in the Rust documentation [here](https://doc.rust-lang.org/book/ch10-01-syntax.html).
+
+For example, you may need functions that add `u8` and `u32` integers. Instead of having to write two functions that do the same thing, you can write one function that works for both using generics.
+
+```rust noplaypen
+fn add<T>(a : T, b : T) -> T {
+  return a + b
+}
+```
```

Comment:
We don't allow printing of structs afaik so there may be no operation which supports this.

Reply:
That is true. There's also a bug with println currently where parameters do not print properly since we try to verify the form of println's parameters before ssa does its optimizations rather than after (see #1035).

Not sure what we could replace this example with then while avoiding introducing struct or numeric generics until later. Perhaps just the classic identity function or return first/second:
```rs
fn id<T>(x: T) -> T  {
    x
}

fn return_second<A, B>(x: A, y: B) -> B {
    y
}
```